### PR TITLE
Unit test course component

### PR DESCRIPTION
**Pull request by lucy811**
1.Fixed unit test errors in other components.
2.Wrote unit tests in courses.component.spec.ts and courses-add.component.spec.ts
unit_test

3. npm run v-lint-fix
v-lint

### CODE REVIEW

**Comment by lmmrssa** [comment]
@lucy811 could you please rebase the branch with master. We had fixed the travis break. So your test might build after that.

File: src/app/nation/nation.component.spec.ts [hunk 1]
```diff
@@ -1,23 +1,26 @@
 import { async, ComponentFixture, TestBed } from '@angular/core/testing';
-import { AlertsDeleteComponent } from '../shared/alerts/alerts-delete.component';
 import { NationComponent } from './nation.component';
 import { HttpClientModule } from '@angular/common/http';
 import { ReactiveFormsModule, FormsModule } from '@angular/forms';
+import { MaterialModule } from '../shared/material.module';
 import { FormErrorMessagesComponent } from '../shared/form-error-messages.component';
+import { Jsonp, JsonpModule } from '@angular/http';
```
Comment (lmmrssa) [inline src/app/nation/nation.component.spec.ts]: we are trying to replace these deprecated elements from newer ones. So could you explore for @angular/common/http equivalent elements

**Comment by lucy811** [comment]
1. I have fixed new unit test errors in other components after rebased the branch with master.
2. I have deleted Jsonp and JsonpModule in nation.component.spec.ts

File: src/app/courses/add-courses/courses-add.component.spec.ts [hunk 2]
```diff
@@ -1,33 +1,97 @@
...
+  // test onSubmit()
+  it('should onSubmit', () => {
+    component.onSubmit();
+    fixture.whenStable().then(() => {
+      fixture.detectChanges();
+      expect(statusElement2.textContent).toBe('This field is required');
+    });
+  });
+
+  // test addCourse()
```
Comment (nirojdyola) [inline src/app/courses/add-courses/courses-add.component.spec.ts]: unnecessary comments can be removed.Could you please check it once ?

File: src/app/resources/resources.component.spec.ts [hunk 3]
```diff
@@ -49,16 +50,19 @@ describe('ResourcesComponent', () => {
   });
 
   // test getRating()
```
Comment (nirojdyola) [inline src/app/resources/resources.component.spec.ts]: unncessary comments can be removed. Could you please check it once ?

File: src/app/users/users.component.spec.ts [hunk 4]
```diff
@@ -94,10 +94,9 @@ describe('Users', () => {
 
     it('Should make a PUT request to CouchDB with role deleted', () => {
       const { comp, couchService } = setup(),
-        testEvent = { stopPropagation: () => { } },
         initSpy = spyOn(comp, 'initializeData').and.callFake(() => { } ),
         couchSpy = spyOn(couchService, 'put').and.returnValue(Promise.resolve({}));
-      comp.deleteRole({ name: 'Test', roles: [ 'one', 'two', 'three' ] }, 1, testEvent);
+      comp.deleteRole({ name: 'Test', roles: [ 'one', 'two', 'three' ] }, 1 );
```
Comment (nirojdyola) [inline src/app/users/users.component.spec.ts]: extra space after 1. Could you please remove it ?

**Comment by lucy811** [comment]
@nirojdyola I have deleted unnecessary comments and extra space.

**Comment by lucy811** [comment]
Added imports:[RouterTestingModule.withRoutes([{ path: 'courses', component: CoursesAddComponent}])] in courses-add.component.spec.ts in order to spyon router in test addCourse().

**Comment by paulbert** [comment]
Hi @lucy811 it worked to simply update the `debugOperator` in `app.component.spec.ts` so that is fixed.  There is one other thing for you to do to update the tests to the changes on the `master`: every time a spy on `CouchService` returned a `Promise.resolve` you will have to:
1. `import { of } from 'rxjs/observable/of';` at the top of the spec file (once per file)
2. Replace `Promise.resolve` with `of` using the same object as the parameter of the function

Right now the tests for me are not completing, but once this is changed it should work.  Let us know if you run into any other issues!

**Comment by lucy811** [comment]
1. Imported { of } from 'rxjs/observable/of' in each .spec file.
2. Replaced Promise.resolve with of
3. Replaced Promise.reject with Rx.Observable.throw
4. Used 'ng lint --fix' to check errors

**Pull request merged**